### DISCUSSION TITLE
Google软件工程之工具篇 · BMPI

### DISCUSSION THREAD

**Discussion by giscus** [bmpi-dev/bmpi.dev#122] [Announcements]
# Google软件工程之工具篇 · 构建我的被动收入

本文是《Software Engineering at Google》的读书笔记，同时会穿插分享我对软件工程的理解。本文主要介绍软件工程的相关工具，主要包括版本控制、分支管理、代码搜索、构建系统、代码静态化分析、依赖管理与CI/CD等。

https://www.bmpi.dev/dev/software-engineering-at-google/tool/

**Comment by madawei2699** [comment]
各种分支管理策略的优势和劣势👇

**Reply by madawei2699** [reply]
![image](https://user-images.githubusercontent.com/2446612/193441795-48514929-cc3b-4ee5-a36d-0d2456687547.jpeg)
